Add new device support for lede

wackejohn opened the pull request:
Added  RTNAS device support for lede.
Bleow is  the RTNAS hardware:

Soc :Marvell armada370 ARMv7 1.2G
Switch: Marvell 88E6171R 100/1000M 5 Port
DDR3 1G or 512MB
NAND FLASH 4GB
SATA2: 2 Port and 12V power Port: 2 socket
Minipci-e: 2 slot (include 4G LTE support)
SIM : 1 socket
GPIO .......

mkresin commented:
Please have a look at the [pull request template draft](https://github.com/bobafetthotmail/source/blob/c7cfdff4fa40f53511b8c8a4bd2e60aa980dc6fd/.github/pull_request_template) and fix your commit title and message.

Please merge your commits into one, add the hardware description and some informations on how to install LEDE to the device to the commit message. Have a look at https://github.com/lede-project/source/commit/fd62fa752bbff9ae5277098152e7960a14d241e1 a good example how it could look like.

Please use lower case partition names in the DTS. Have a look at the other dts files in LEDE for examples.

wackejohn commented:
Thanks for reply,I'm not a developer,I'll try to fix them.
And this device need kernel model to make 88e6171 switch working,how to make a patch to change the kernel config(e.g. config-4.4 file)?

LipkeGu commented:
Simply add the CONFIG_********* variable to the config-44 file

wackejohn commented:
I forgot to say that the uboot of this device specified a kernel command line and we can't change it(because the uboot will overwrite it every boot).
    So to make lede booting,we need specify a kernel command line while compile the kernel.But this command line can't simply add to the kernel config file,or it will make other device not booting.
    Another problem is lede can't mount the ubifs overylay fs (exroot) properly while booting (use squeshfs+ubifs overylay file system),I fix this by add a fstab config file to the rootfs.But this file also can't add to pull request too.
   Any idea?Thanks.

bobafetthotmail commented:
>I forgot to say that the uboot of this device specified a kernel command line and we can't change it

What is the command line passed? 

the mvebu kernel config has CONFIG_MANGLE_BOOTARGS that should already remove some entries form kernel command line. See this patch for what that does.
https://github.com/lede-project/source/blob/4f272dd0323151a1390a8675bafebead0e2a69ad/target/linux/generic/patches-4.4/995-mangle_bootargs.patch

> Another problem is lede can't mount the ubifs overylay fs

You tried calling the root filesystem partition (is it the one called "rootfs"?) **ubi** in the dtb file?

wackejohn commented:
> What is the command line passed?

**The uboot bootargs line:**
`bootargs=console=ttyS0,115200 ubi.mtd=5 root=ubi0:rootfs ro rootfstype=ubifs`

**My kernel config command line:**
`CONFIG_CMDLINE="console=ttyS0,115200 ubi.mtd=5 root=/dev/mtdblock5 rootfstype=squashfs"`

> You tried calling the root filesystem partition (is it the one called "rootfs"?) ubi in the dtb file?

Yes,in the dtb file the partition called "rootfs",but when i use the squashfs and ubifs overly,this partition will be "ubi".

**Below is my device boot log:**

[    0.908750] 7 ofpart partitions found on MTD device pxa3xx_nand-0
[    0.914872] Creating 7 MTD partitions on "pxa3xx_nand-0":
[    0.920321] 0x000000000000-0x000000400000 : "U-Boot"
[    0.926322] 0x000000400000-0x000000800000 : "UBoot_env"
[    0.932660] 0x000000800000-0x000000c00000 : "Vendor"
[    0.938678] 0x000000c00000-0x000001800000 : "Unused"
[    0.944662] 0x000001800000-0x000001c00000 : "kernel"
[    0.950833] 0x000001c00000-0x000040000000 : "ubi"
[    0.956809] 0x000040000000-0x0000fbc00000 : "syscfg"

**And my fstab config file:**

config global
	option anon_swap '0'
	option anon_mount '0'
	option auto_swap '1'
	option auto_mount '1'
	option delay_root '5'
	option check_fs '0'
	
config mount 
	option target '/overlay' 
	option device '/dev/ubi0_1'
	option options 'rw,sync'
	option enabled '1' 
	option fstype 'ubifs'

wackejohn commented:
Is there anyone  can help me?

bobafetthotmail commented:
>My kernel config command line:
>CONFIG_CMDLINE="console=ttyS0,115200 ubi.mtd=5 root=/dev/mtdblock5 rootfstype=squashfs"

I think the "ubi.mtd=5 root=/dev/mtdblock5 rootfstype=squashfs" is not needed. 
LEDE will automatically find and use the partition called "ubi" without need of any option.

See this patch https://github.com/lede-project/source/blob/4f272dd0323151a1390a8675bafebead0e2a69ad/target/linux/generic/patches-4.4/490-ubi-auto-attach-mtd-device-named-ubi-or-data-on-boot.patch

Try writing "ubi" instead of "rootfs"  in the dtb and use only "CONFIG_CMDLINE="console=ttyS0,115200", no root filesystem options.

wackejohn commented:
> I think the "ubi.mtd=5 root=/dev/mtdblock5 rootfstype=squashfs" is not needed.
LEDE will automatically find and use the partition called "ubi" without need of any option.

**But this device bootargs line was written to the uboot by the oem, on every boot the command line will be overywrited by the uboot of this device...the only thing we can do is overwrite the command line in the kernel.**

**And if I delete the fstab config file,LEDE can't mount the overlay fs properly,and also can't save any config,will lose all the configuartion after a reboot.**

**Below is my another device boot log:**
[   19.250417] mount_root: failed to launch kmodloader from internal overlay
[   19.792115] UBIFS (ubi0:1): background thread "ubifs_bgt0_1" started, PID 1030
[   19.878912] UBIFS (ubi0:1): recovery needed
[   19.929346] random: procd: uninitialized urandom read (4 bytes read, 46 bits of entropy available)
[   20.047116] UBIFS (ubi0:1): recovery completed
[   20.100426] UBIFS (ubi0:1): UBIFS: mounted UBI device 0, volume 1, name "rootfs_data"
[   20.194284] UBIFS (ubi0:1): LEB size: 15872 bytes (15 KiB), min./max. I/O unit sizes: 512 bytes/512 bytes
[   20.308955] UBIFS (ubi0:1): FS size: 16840192 bytes (16 MiB, 1061 LEBs), journal size 841216 bytes (0 MiB, 53 LEBs)
[   20.434047] UBIFS (ubi0:1): reserved for root: 795403 bytes (776 KiB)
[   20.511231] UBIFS (ubi0:1): media format: w4/r0 (latest is w4/r0), UUID 68E7489F-C3F2-4358-B2C3-B3401EE69DE2, small LPT model
[   20.647399] UBIFS (ubi0:1): full atime support is enabled.
[   20.715442] block: attempting to load /tmp/ubifs_cfg/upper/etc/config/fstab
[   20.801148] block: extroot: not configured
[   20.850333] UBIFS (ubi0:1): un-mount UBI device 0
[   20.906743] UBIFS (ubi0:1): background thread "ubifs_bgt0_1" stops
[   20.992068] UBIFS (ubi0:1): background thread "ubifs_bgt0_1" started, PID 1032
[   21.080735] UBIFS (ubi0:1): UBIFS: mounted UBI device 0, volume 1, name "rootfs_data"
[   21.174530] UBIFS (ubi0:1): LEB size: 15872 bytes (15 KiB), min./max. I/O unit sizes: 512 bytes/512 bytes
[   21.289205] UBIFS (ubi0:1): FS size: 16840192 bytes (16 MiB, 1061 LEBs), journal size 841216 bytes (0 MiB, 53 LEBs)
[   21.414296] UBIFS (ubi0:1): reserved for root: 795403 bytes (776 KiB)
[   21.491483] UBIFS (ubi0:1): media format: w4/r0 (latest is w4/r0), UUID 68E7489F-C3F2-4358-B2C3-B3401EE69DE2, small LPT model
[   21.627615] UBIFS (ubi0:1): full atime support is enabled.
[   21.892284] block: attempting to load /tmp/ubifs_cfg/upper/etc/config/fstab
[   21.977844] block: extroot: not configured
[   22.028786] mount_root: overlay filesystem has not been fully initialized yet
[   22.312686] mount_root: switching to ubifs overlay

bobafetthotmail commented:
On 02/16/2017 03:08 PM, wackejohn wrote:

I think the "ubi.mtd=5 root=/dev/mtdblock5 rootfstype=squashfs" is not needed.
LEDE will automatically find and use the partition called "ubi" without need of any option.

But this device bootargs line was written to the uboot by the oem, on every boot the command line will be overywrited by the uboot of this device...the only thing we can do is overwrite the command line in the kernel.

As I said, the mvebu in LEDE has a patch that should rename the bootargs passed from uboot into something else (it is called "mangle bootargs") so it is not used, did you actually try doing what I said?

wackejohn commented:
> As I said, the mvebu in LEDE has a patch that should rename the bootargs passed from uboot into something else (it is called "mangle bootargs") so it is not used, did you actually try doing what I said?

I removed the root filesystem options and renamed "rootfs" to "ubi" in dtb file. LEDE can found the ubi partition successfully.(I tested on another kirkwood device,will test  this device later).But after mount the rootfs(squashfs),LEDE can't mount the rootfs_data(ubifs) successfully,and will lose the configuartion after reboot.Is there anyway that make LEDE mount the rootfs_data(ubifs) without fstab file? Thanks.

bobafetthotmail commented:
what is the error?

Did you try writing **umount /overlay && firstboot && reboot** in LEDE terminal? this should reformat the rootfs_data partition.

wackejohn commented:
> what is the error?

kernel log like this:

[   22.068974] mount_root: overlay filesystem has not been fully initialized yet
[   22.347933] mount_root: switching to ubifs overlay

and after reboot still this:

[   22.035510] mount_root: overlay filesystem has not been fully initialized yet
[   22.313017] mount_root: switching to ubifs overlay


> Did you try writing umount /overlay && firstboot && reboot in LEDE terminal? this should reformat the rootfs_data partition.

root@LEDE:/# umount /overlay && firstboot && reboot
This will erase all settings and remove any installed packages. Are you sure? [N/y]
y
/dev/ubi0_1 is n[   77.409798] ubi0 error: ubi_open_volume: cannot open device 0, volume 1, error -16
ot mounted
/dev/ubi0_1 will be erased on next mount
writing /dev/ubi0_1 failed: Bad file descriptor

mkresin commented:
The changes I've requested 23 days ago are still not done. Do you plan to fix the stuff I asked for? If not, please close the PR till you have the time to take care of the requested changes.
